## Methods {#methods}

### Sampling

Premetamorphic tadpoles (TK stages I-VI) were exposed at 21 °C for 48 h to concentrations of T3 (Sigma-Aldrich, Oakville, ON; Catalog #T2752, CAS 55-06-1), T4 (Sigma, Catalog #T2501, CAS 6106-07-6), or E2 (Sigma, Catalog #E4389, PubChem Substance ID: 329799056). 
Five tadpoles per exposure condition were exposed to either 10 nM T3 in 40nM NaOH, 50 nM T4 in 50nM NaOH exposure, 10 nM E2 in dechlorinated water, or their respective vehicle control. 
The concentrations chosen were based upon observed physiological and environmental relevance [@doi:10.22621/Maher et al., 2016]
After completion of the experimental exposures, tadpoles were euthanized in 0.1% buffered tricaine methanesulfonate (TMS, Aqua Life, Syndel Laboratories, Nanaimo, BC, Canada) and the TF and rostrum were dissected, preserved in RNAlater solution (Ambion, Foster City, CA, USA), and stored at -20 °C. 
The rostrum was halved and sub-dessected to isolate the olfactory sac containing the OE. 
The OE and TF from each sample animal were then combined in 300 μL TRIzol and mechanically disrupted in a Retsch MM301 Mixer Mill (Thermo Fisher Scientific, Ottawa, Canada) at 25 Hz for two 3-min intervals, separated by a 180° rotation of the samples. 
After pelleting insoluble material with centrifugation at 12,000 ×g for 10 min at 4 °C, the supernatant was transferred to a new RNAse-free tube. 
RNA was extracted and washed using chloroform, isopropanol, and ethanol treatments and subsequently dissolved in 30 μL diethyl pyrocarbonate-treated water (Sigma-Aldrich) and stored at −80 °C.


### 16S rRNA Amplicon Sequencing

We performed 16S ribosomal RNA (rRNA) sequencing on 38 tadpole and frog samples, 5 control swabs from gloves and swabs, and 6 control swabs from tank biofilm and water.
A set of positive controls consisting of predefined communities, and negative controls were also included as part of the sequencing experiment. 
The 16S rDNA was amplified using the following primer pair. 
Forward: Klin16SF: 5'-TCGTCGGCAGCGTCAGATGTGTATAAGAGACAGAYTGGGYDTAAAGNG-3' and reverse Klin16SR: 5'-GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAGTACNVGGGTATCTAATCC-3'
After amplification, the amplicons are prepared for sequencing following the Illumina 16S Metagenomic Sequencing Library Preparation guide (Ref. Illumina. 2017). 
The success of library preparation was verified by agarose gels. 
Equimolar amount of all samples with an additional positive and negative control was pooled and sequenced in four independent runs with the Illumina MiSeq platform using the MiSeq Reagent Kit V2 (Illumina), producing 251 bases paired end reads. 
The sequence data is available in NCBI SRA database under biosample ID 123456789. 

### 16S rRNA Amplicon Analysis 

Raw sequencing reads were demultiplexed and processing within QIIME2 (Ver. 2018.4) [@doi:10.1038/s41587-019-0209-9] following the standard ASV identification workflow. 
Amplicon sequence variants (ASVs) identified using QIIME2 submodule DADA2 [@doi:10.1038/nmeth.3869]. 
Taxonomic assignment of ASVs was done using QIIME2 VSEARCH submodule against the Silva 16s rRNA database (ver. 132, 99%;) [@doi:10.1093/nar/gks1219].  
Alpha diversity of microbiomes was calculated using the Shannon-Wiener Diversity Index. 
Beta diversity was measured using the weighted-unifrac distance and visualized using PCoA. 
Taxa correlation with life stage was identified using pearson correlation. 

### Meta-transcriptomic Sequencing

The RNA quality and quantity were verified by a Nanodrop spectrophotometer (Thermo Scientific NanoDrop One spectrophotometer, Thermo Scientific). 
The integrity of isolated RNA was analyzed using a Bioanalyzer 2100 (Agilent Technologies), and samples with RNA integrity number (RIN) of > 7 were used for RNA-Seq analyses (5 biological replicates per treatment). 
RNA samples were shipped on dry ice to Canada's Michael Smith Genome Sciences Centre (GSC, BC Cancer Research, Canada), where strand-specific mRNA libraries were constructed and sequenced using Illumina HiSeq 2500 (paired-end platform generating 2 x 75 base pair reads for each sample).
The sequence data is available in NCBI SRA database under biosample ID 123456789. 

### Meta-transcriptiomic Analysis

Our RNA-Seq analysis pipeline follows the schematic presented in figure (@fig:methodology_overview). 
Reads quality was assessed using FastQC to ensure high read quality in all samples. 
Using TransAbyss, the fastq reads generated from each sample was assembled denovo into a reference transcriptome library representing each transcript that could be assembled given the paired-end reads supplied. 
We created a complete reference transcriptome assembled using TransAbyss All reference transcriptome libraries were then merged and filtered to remove duplicated sequences, and this merged reference transcriptome was filtered through a large library of microbial RefSeq sequences using Kraken2 to create a library of microbial transcript sequences that were found in one or more of the sequenced samples. 

Kraken2 is a taxonomic classification system that relies on exact k-mer matches instead of percent identity and coverage. 
Kraken2 matches the query sequences to the lowest common ancestor that is matched within the selected database. 
Kraken2 will assign a taxonomy if the number of identical k-mer matches passes a threshold percentage of all k-mers of a given sequence, as set by the user. 
The optimal threshold for taxonomic assignment in this dataset was determined to be a 40% confidence threshold of k-mer matches (Fig. @fig:kmer_threshold).

The Kraken2 database used to filter for microbial transcripts includes the following RefSeq sequences:
-Archaea: RefSeq complete archaeal genomes/proteins
-Bacteria: RefSeq complete bacterial genomes/proteins
-Viral: RefSeq complete viral genomes/proteins
-Human: GRCh38 human genome/proteins
-Fungi: RefSeq complete fungal genomes/proteins
-Plant: RefSeq complete plant genomes/proteins
-Protozoa: RefSeq complete protozoan genomes/proteins

Human RefSeq sequences were included in the Kraken2 filter in order identify human contamination and separate these transcripts later in the microbial RNA-Seq pipeline.

We used this new filtered list of microbial transcripts as our microbial reference transcriptome and re-mapped the raw reads from each sequenced sample to this new reference using Burrow-Wheeler Alignment tool. 
The mapped reads were the assembled into transcripts and counted for differential expression analysis using DESeq2. 
We then extracted two lists of sequences for each WEC exposure condition and tissue type: one list containing all transcripts that were significantly represented in the majority of samples within a given exposure and control set (cpm > 0.1), and one list containing all transcripts that were significantly differentially expressed with a specific exposure condition. 
These two lists were then fed through the kraken2 databases consisting of each of the RefSeq libraries above, which will assign a taxonomy to a transcript at a match threshold.


![**RNASeq methodology overview**. figure description. ](images/method_overview.png){#fig:methodology_overview width="6.5in"}
![**K-mer threshold for taxonomy identification**. Excluding counts generated using a 10 to 30 % confidence intervals for kraken2 mapping creates a linear correlation (R2 > 0.95) between percent confidence and number of bacterial.](images/kmer_threshold.png){#fig:kmer_threshold width="6.5in"}
 