## Results {#results}

### Characterization of Lithobates catesbeianus skin bacterial microbiome across life stages. 

Overall, seven phyla were detected at a high abundance and represented >95% of all bullfrog microbiomes (Fig. @phylum_combined). 
Proteobacteria were the most dominant in tadpoles (median 27%) followed by Fusobacteria (median 12%), but both of their relative abundance decreased with life stage. 
At the froglet stage, Verrucomicrobia (median 89%) became the most dominant phyla. 

![**Relative Abundance of 16S Microbiomes at the Bacterial Phylum Level**. The relative abundance of the top 10 bacterial phyla present across all samples represents >99% of all reads sequenced. The  tadpole skin microbiome (labelled 1-23) consisted of predominately Proteobacteria (Purple). However, as life stage progresses, froglets are dominated by Verrucomicrobia. The relative distribution of phyla in the positive control (labelled POS) is as expected. Sterile swabs (labelled SSC) and negative controls (labelled NSC) have <2k reads on average. ](images/phylum_label.png){#fig:phylum_combined width="6.5in"}

We further explored the difference between the lifestages using an alpha diversity measure via the Shannon-Wiener index. 
Tadpoles exhited the greatest alpha diversity, decreasing as it progresses to froglets (Fig. @alpha_diversity). 
Using unweighted UniFrac distance, we showed the separation of microbiomes associated with each life stage. 
We captured this separate using a 2-axis PCoA, demostrating a distinct separation of froglet and tadpoles, with samples in different stages of metamorphosis scattered in between (Fig. @bray_nmds, Fig. @bray_nmds_cluster). 
Frog sex, mass and length of the tadpole had no impact on the community composition (Fig. @sex_mass_length)

![**Shannon Diversity Index of The Microbiomes Comparing Different Conditions**. ](images/alpha_diversity.png){#fig:alpha_diversity width="6.5in"}
![**NMDS of the micorbiomes**. ](images/bray_nmds.png){#fig:bray_nmds width="6.5in"}
![**NMDS of the micorbiomes**. ](images/bray_nmds_cluster.png){#fig:bray_nmds_cluster width="6.5in"}

![**Community composition did not correlate with mass, size, or sex**. ](images/sex_mass_length.png){#fig:sex_mass_length width="6.5in"}


### The effect of T3 and T4 Exposure on the Lithobates catesbeianus skin microbiome. 

The skin microbiome were analysed based on 40 sequenced samples exposed to T3 or T4 (across premet, promet and froglets life stages). 

The skin microbiota of the T3 and T4 exposed samples were dominated by proteobacteria and bacteriodetes. 
The exposure of T4 but not T3 led to an increase in alpha diversity measured by both Shannon H and Simpson index with an enrichment in Caulobacter henricii and an unclassified chlamydiaceae with T4 exposure and an unclassified enterobacteriaceae with T3 exposure [Fig. @alpha_diversity_t]. 
PCoA of UniFrac distances based on normalised OTU abundances did not show any clustering between the samples (Fig. @nmds_t, Fig. @nmds_t_cluster). 

![**Shannon Diversity Index of The T3/T4 Microbiomes Comparing Different Conditions**. ](images/alpha_diversity_t.png){#fig:alpha_diversity_t width="6.5in"}
![**PCoA of the T3/T4 treated micorbiomes**. ](images/unifrac_pcoa_t.png){#fig:nmds_t width="6.5in"}
![**PCoA of the T3/T4 treated micorbiomes**. ](images/unifrac_pcoa_t.png){#fig:nmds_t_cluster width="6.5in"}


#### Meta-transcriptomic analysis of E2, T3, and T4 Exposure

From the counts generated from the tank water data set, primarily proteobacteria and Bacteroidetes were counted, specifically bacteria from the flavobacterium genus.
Assigning bacterial taxonomies to de novo assembled transcripts from the TF and OE of tadpoles from the E2, T4, and T3 exposure sets revealed the presence of mostly bacteria from the proteobacteria phylum, with a smaller portion of bacterial transcripts mapping to gram positive phyla actinobacteria and firmicutes in both tissues (table 2, table 4). 
The OE had more bacterial transcripts that were significantly represented compared to the TF. 
Agrobacterium and Pseudomonas species exhibited the most varieties of transcripts of all present species.
TFT3 exposure resulted in the differential expression of T226 16S ribosomal RNA that is assigned to Micrococcus luteus (Table 3). 
The M. luteus transcript exhibits a 6.5-fold change in transcript abundance (table 9). The M. luteus transcript exhibits similarity to an M. luteus 16S rRNA gene (Accession MT611279). 
No other treatments results in differential expression of other bacterial transcripts.
Viral taxonomies were also assigned to assembled transcripts. 
There are fewer viral transcript assignments than bacterial in both tissues (table 5, table 7). 
The distribution of viral families assigned transcripts is very similar between tissues, with the majority belonging to Iridoviridae, followed by Adenoviridae, and Siphoviridae. Three viral species were commonly assigned to transcripts from both tissues: proteus phage VB PmiS-Isfahan, Frog virus 3, and human mastadenovirus C.
In the TF, T4 exposure results in the differential expressed of a single transcript assigned to the Frog virus 3 species (Table 7) with a 5.5-fold change of transcript abundance (table 13). 
Through a blastn query, this transcript is confirmed to encode for a hypothetical protein FV3gorf78L from the Frog virus 3 genome which contains an RNA recognition motif. 
T3 and E2 exposure did not result in the differential expression of any viral assigned transcripts in the TF.
In the OE, both T4 and T3 exposure results in the differential expression of viral assigned transcripts from Proteus phage VB_PmiS-Isfahan, however, these are annotated with the human mitochondrial genome and are likely not viral transcripts (table 14-17).
No significant results were found from filtering with the other RefSeq databases with Kraken2.
