## Discussion {#discussion}

This study characterized the skin microbiome of North American Xenopus frogs as it metamorphosizes naturally as well as when exposed to endocrine disrupting compounds (EDCs).
These data combined gave us an overview of 1) the core frog skin microbiome, 2) variations of the skin microbiome through tadpole development and metamorphosis, and 3) potential effect of EDCs on the skin microbiome through development.

#### The Tadpole Skin Microbiome is Unique and Shares a Core Composition that Changes With Natural Metamorphosis
Analysis of the tadpole and froglet microbiomes revealed a Proteobacteria domination in their core microbiome. 
However, this core composition shifts to a Verrucomicrobia dominance as the tadpoles naturally metamophosizes to froglets.
This increase in Verrucomicrobia is composed of an increase in the order Verrucomicrobiales.
There was not enough resolution to classify these ASVs further.
These predominant phyla found in the frog skin microbiomes are consistent with the skin microbiomes of other marine mammals in both the wild and captive (For example: Killer whales [@doi:10.1038/s41598-017-00457-5], captive dolphins [@doi:10.1038/ncomms10516]). 

#### EDC Exposure Did Not Led to Any Changes in the Frog's Microbiome.
The 16S sequencing results demonstrated no significant differences between the microbiomes of E2, T3 and T4 exposed tadpoles. However, the assigning of bacterial taxonomies to expressed transcripts reveals a very slight response with T3 exposure in both tissues and T4 exposure did not result in the differential expression of any bacteria-assigned transcripts (table 3, table 5).

Assigning bacterial taxonomies to assemble transcripts revealed similar bacteria phyla as the 16S sequencing result. 
Both techniques capture mostly bacteria belonging to the phylum proteobacteria, with firmicutes captured as well. 
16S sequencing uniquely captures the presence of Bacteroidetes and fusobacteria, where RNA-Seq methods capture the presence of actinobacteria.

The microbial RNA-Seq only detected a difference in bacteria-assigned transcript response with the T3 exposure (table 3, table 5). 
The transcript assigned to C. botulinum transcript exhibits significant abundance in all exposure sets and appears to be responsive to T3 exposure in both tissues. 
Since the blastn annotation of this gene returns X. laevis rRNA as the top hits, it is unclear whether this transcript is really from a bacterial rRNA gene and has been misannotated in NCBIâ€™s database or is a frog gene and happens to have at least 40% of its k-mers being identical matches to C. clostridium. 
The transcript assigned to M. luteus is confirmed to be a segment of the 16S rRNA gene from that bacterial species with the blastn annotation. 
This bacteria is commonly found in soil, dust, water and air, and as part of the normal microbiota of the mammalian skin [@doi:10.1007/s00248-003-2016-5].

The microbial RNA-Seq method captured a variety of virus-assigned transcripts that cannot be capture with 16S sequencing methods (table 6, table 8). 
In this dataset, there is likely a high false positive from human contamination due to retroviruses in the virus database which may have gene segments that match the human genome. 
The presence of mostly ranavirus represented in these data supports that this method is actually capturing the presence of viral transcripts (table 7). 
The differential expression of the frog virus 3 annotated transcript is the most interesting result from this investigation, although there is not much information available about this gene. 
The transcript is confirmed to be a viral-protein encoding gene by blastn annotation and exhibits a 5.5-fold change with T3 exposure in the TF (table 13). 
Frog virus 3 can be commonly isolated from healthy frogs and its role in frog disease is unknown [@doi:10.1016/B978-012362675-2/50012-4].

Three different transcripts assigned to Proteus phage VB_PmiS-Isfahan are differentially expressed in either the OET3 or OET4. 
All these transcripts exhibit a roughly 0.5-fold change with either T3 or T4 exposure and are present in quite high abundance of 600 to 1800 median counts (table 14 - 17). 
These transcripts are likely false positives as they are annotated to be from human mitochondrial genomes following a blastn query. 
A transcript assigned to Human endogenous retrovirus K is also differentially expressed in the OE, with a 3-fold change in transcript abundance with T3 exposure (table 18). 
This transcript is in much lower abundance and although it was determined to be the HERV-K gene from this retrovirus, it is still likely that this is a false positive from human contamination.
