![**RNASeq methodology overview**. figure description. ](images/method_overview.png){#fig:methodology_overview width="6.5in"}

![**K-mer threshold for taxonomy identification**. Excluding counts generated using a 10 to 30% confidence intervals for kraken2 mapping creates a linear correlation (R2 > 0.95) between percent confidence and number of bacterial.](images/kmer_threshold.png){#fig:kmer_threshold width="6.5in"}

![**Relative Abundance of 16S Microbiomes at the Bacterial Phylum Level**. The relative abundance of the top 10 bacterial phyla present across all samples represents >99% of all reads sequenced. The  tadpole skin microbiome (labelled 1-23) consisted of predominately Proteobacteria (Purple). However, as life stage progresses, froglets are dominated by Verrucomicrobia. The relative distribution of phyla in the positive control (labelled POS) is as expected. Sterile swabs (labelled SSC) and negative controls (labelled NSC) have <2k reads on average. ](images/phylum_label.png){#fig:phylum_combined width="6.5in"}

![**Shannon Diversity Index of The Microbiomes Comparing Different Conditions**. ](images/alpha_diversity.png){#fig:alpha_diversity width="6.5in"}

![**NMDS of the micorbiomes**. ](images/bray_nmds.png){#fig:bray_nmds width="6.5in"}

![**NMDS of the micorbiomes**. ](images/bray_nmds_cluster.png){#fig:bray_nmds_cluster width="6.5in"}

![**Community composition did not correlate with mass, size, or sex**. ](images/sex_mass_length.png){#fig:sex_mass_length width="6.5in"}

![**Shannon Diversity Index of The T3/T4 Microbiomes Comparing Different Conditions**. ](images/alpha_diversity_t.png){#fig:alpha_diversity_t width="6.5in"}

![**PCoA of the T3/T4 treated micorbiomes**. ](images/nmds_t.png){#fig:nmds_t width="6.5in"}

![**PCoA of the T3/T4 treated micorbiomes**. ](images/nmds_t_cluster.png){#fig:nmds_t_cluster width="6.5in"}
